### PR TITLE
Allow Multiple Air Loops to One Thermal Zone - Part 2 and Fix Zone Air Mass Balance

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
 Followup work to complete and clean up #6248. This pull request is focused on the simulation side of multiple airloops - getting rid of the one-return-node/one-airloopnum per zone mindset and revising the whole zone air mass balance to split thing up across multiple return nodes in a zone.

- [x] Address #6264 Followups from Multiple Airloops Part 1 (see checklist there)
- [x] Address #6273 DOATToUnitVentilator node connection errors (never released this way, error introduced in #6248)
- [x] Address #6219 Zone air mass balance is broken.
- [x] Finish implementing options to specify the return air node for heat gain from lights, airflow windows, and refrigeration cases.
- [x] Finish implementing allocation of return air flow to multiple airloops serving the same zone.
- [x] Engineering ref changes.

#### 6 files with big diffs *as of 2017-09-21*
5ZoneAirCooledWithDOASAirLoop and DOASDXCOIL_wADPBFMethod_NoReturnPath - big table diffs, because the outdoor air summary table is correct now (previously zeros)

5ZoneAirCooled_ZoneAirMassFlowBalance and 5ZoneAirCooled_ZoneAirMassFlowBalance_Pressurized - big eso diffs etc because zone air mass conservation is fixed here

DOAToWaterToAirHPInlet - These diffs should go away once #6333 drops in.

VSHeatPumpWaterHeater - big table and eio diffs because there is one less warmup day now *as of 09-25 there are now a few more warmup days - but no code changes here just merging in develop*

#### 3 files with err diffs *as of 2017-09-21*
DOAToUnitVentilator - Node connection errors are gone.

HospitalLowEnergy - One fewer SolveAirLoopControllers warning for DOAS1 and DOAS5.

VariableRefrigerantFlow_FluidTCtrl_HR_5Zone - New unbalanced air flow warnings - should go away when #6301 is fixed.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Comment by rraustad** [comment]
What's the status here. I don't see anything in the diff's that stand out.

**Comment by mjwitte** [comment]
Still wrestling with this.  And have some other parts to implement.  The thing is, at this point it should be zero diffs.  But finding interesting things that are contributing to diffs - like [this](https://github.com/NREL/EnergyPlus/compare/ResetNodeDataEarlier) which is likely responsible for ReverseDD failures in the two data center files.

**Comment by rraustad** [comment]
I thought ResetNodeData was already called at BeginEnvironment. In your "reset earlier" branch, I see you added another instead of moving the existing call. That seems wrong (should only call once), but of course your the one looking at this. 

I also see that the plenum does it's own node reset. That seems redundant unless node data isn't read in first.

**Comment by mjwitte** [comment]
Probably correct to move ResetNodeData instead of calling it twice.  The place it's called currently (in develop) is too late for the IT equipment which uses a supply node temp to calculate it's power consumption.  It showed up because the results coming out of SetUpSimulation (which runs everything for a few time steps) was different in this branch (because the airloop nums aren't all intialized yet) but that *shouldn't* matter, because in theory everything gets reset on begin environment.  Follow?  Sorry, confusing.

The plenum wants to start at 20C like the zones do.  But ResetNodeData sets everything to 0C.  For the IT equipment, I *could* add a begin environment init there instead, but I wanted to see if it might change some other files - which it did not.  Still seems easier just to move ResetNodeData.  I should open a PR for that branch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -84,7 +84,7 @@ if( BUILD_TESTING )
       set( ARCH_FLAG "" )
     endif()
 
-    set( REGRESSION_BASELINE_SHA "" CACHE STRING "SHA of baseline comparison" )
+    set( REGRESSION_BASELINE_SHA "73377e2" CACHE STRING "SHA of baseline comparison" )
```
Comment (mjwitte) [inline CMakeLists.txt]: @Myoldmopar Did I do this right?  I want to compare this branch  with branch [ResetNodeDataEarlier](https://github.com/NREL/EnergyPlus/tree/ResetNodeDataEarlier)

Comment (Myoldmopar) [reply CMakeLists.txt]: Nah, I'd revert that out.  Change it [here](https://github.com/NREL/EnergyPlus/blob/45a2334aa5db27e9a51dd50d040e5f667a5658d1/.decent_ci.yaml#L6) instead.

**Comment by mjwitte** [comment]
Diffs explanations (as of https://github.com/NREL/EnergyPlus/pull/6272/commits/0f8a85a5843962dd1314d908f685b6bebd02720d )

5ZoneAirCooledWithCoupledInGradeSlab and UnitarySystem_WaterCoils_wMultiSpeedFan - eio warmup convergence diffs and tiny eso diffs

5Zone_Unitary_HXAssistedCoil - `** Warning ** CoilSystem:Cooling:DX:HeatExchangerAssisted "HEAT EXCHANGER ASSISTED COOLING COIL 1" -- Exceeded max iterations error continues` - total error count is 68 now instead of 70, almost all during warmup. No other diffs.

DOAToUnitVentilator - Node connection warnings introduced earlier with this feature are now gone.

HospitalLowEnergy - Warning during warmup is gone `In AirLoopHVAC DOAS2 there is unbalanced exhaust air flow.` All other results have no diffs.  This warning normally is allowed to print only once.  Ran with a version of develop that allowed this warning to always print, and there were only six occurences, all during warmup.

File: doc/engineering-reference/src/integrated-solution-manager/zone-air-mass-flow-conservation.tex [hunk 2]
```diff
@@ -22,35 +22,90 @@ \subsubsection{Overiew}\label{overiew}
...
+
+If there is only one return node and no Zone Return Air Node 1 Flow Rate Basis Nodes, then:
+
+\begin{equation}
+{\dot m_{R,1}} = ReturnFlowSchedule*{\dot m_{R}}
+\end{equation}
+
+\paragraph{Allocation to Multiple Return Nodes}\label{allocation-to-multiple-return-nodes}
+
+If there are multiple return air nodes in the zone, then each return air node is assumed to be paired with on supply air inlet node. The initial flow rate at a given return air node is based on the supply air inlet node for the same airloop and the AirloopHVAC Design Return Air Flow Fraction of Supply Air Flow:
```
Comment (rraustad) [inline doc/engineering-reference/src/integrated-solution-manager/zone-air-mass-flow-conservation.tex]: ... paired with ONE supply air inlet node.

**Comment by rraustad** [comment]
I went throught this again last night. It is admittedly hard to follow the intricate changes necessary for two air loops serving the same zone. Nothing stood out as blatently wrong. The unit test shows that return air flow can be properly managed. Diff's are minimal and explained. Question for HospitalLowEnergy results - if warning is now gone, why were there 6 occurrences during warmup for the special build?

**Comment by mjwitte** [comment]
Not sure about HospitalLowEnergy - that file runs terribly slow in debug.  In fact, the time stamps on those are all 01/01 which isn't even one of the environments for this file.  That says the warnings are happening during SetUpSimulation which really shouldn't be producing warnings at all.  With the new multi-airloop world, the airloop number mapping to zone equipment isn't known yet during SetUpSimulation so that explains why these aren't present in this branch.

Regarding the changes, there are more to come.

**Comment by mjwitte** [comment]
@Nigusse Please look at 5ZoneAirCooled_ZoneAirMassFlowBalance and 5ZoneAirCooled_ZoneAirMassFlowBalance_Pressurized with this branch and see if you agree that this fixes #6219 Zone air mass balance.

**Comment by Myoldmopar** [comment]
My final adventure...I'm going to go ahead and kick off an RC package so it can build overnight and I can feel good about the release package situation.  And I'll get this reviewed along with #6338.

**Comment by Myoldmopar** [comment]
So how coupled are #6338 and this?  I saw the title change on the other one to include a reference to fixes for multiple air loops.  Does that mean these two are basically coupled?  If so, what do you think about pulling that branch into here, or vice versa, to make sure you are capturing the combined effects?  If they aren't that coupled and I'm just reading it wrong, that's fine too.  Things just still feel squishy, so I'm hesitating before investing much more review time.

**Comment by mjwitte** [comment]
This pull request is focused on the simulation side of multiple airloops - getting rid of the one-return-node/one-airloopnum per zone mindset.  And also, revising the whole zone air mass balance to split thing up across multiple return nodes in a zone.  #6338 fixes some issues with the new air terminal sizing inputs - cases that I had not tested before now.  So, they are closely related, yet somewhat separate.  If you'd prefer one combo pull request to review, I can do that too.

**Comment by mjwitte** [comment]
@Myoldmopar Air terminal sizing limitations added to I/O Ref.  Ready for final review here.

**Comment by Myoldmopar** [comment]
Stray unit test failure is unrelated.  Linux failures are the s3 problem.

Doing final code review now.

File: doc/input-output-reference/src/overview/group-design-objects.tex [hunk 3]
```diff
@@ -884,7 +884,7 @@ \subsubsection{Inputs}\label{inputs-012}
...
+\item  \emph{IndoorAirQualityProcedure} means that the program will use the other procedure defined in ASHRAE Standard 62.1-2007 to calculate the amount of outdoor air necessary in order to maintain the levels of indoor air carbon dioxide at or below the setpoint defined in the ZoneControl:ContaminantController object. Appendix A of the ASHRAE 62.1-2010 userâ€™s manual discusses another method for implementing CO2-based DCV in a single zone system. The last two methods of Proportional Control calculate the required outdoor air flow rate which varies in proportion to the percentage of the CO2 signal range and has two choices to calculate occupancy-based outdoor air rate.
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-design-objects.tex:887]: There's still a funny character in here, but I'm assuming there's not enough need to push more commits solely for this.

image

File: doc/input-output-reference/src/overview/group-design-objects.tex [hunk 4]
```diff
@@ -1522,6 +1522,13 @@ \subsection{DesignSpecification:AirTerminal:Sizing}\label{designspecificationair
 
 This object modifies the sizing of an air loop terminal unit. It may be referenced by a ZoneHVAC:AirDistributionUnit or AirTerminal:SingleDuct:Uncontrolled object. The values specified here are applied to the base sizing results from the corresponding Sizing:Zone inputs. Any given DesignSpecification:AirTerminal:Sizing object may be used by multiple terminal units with similar characteristics.
 
+This feature has the following limitations:
+\begin{itemize}
+\item It does not work with VRP sizing.
+\item It does not work with a specified system flow rate in Sizing:System.
+\item It does not work if the base minimum outdoor air flow exceeds the heating or cooling air flow.
+\end{itemize}
```
Comment (Myoldmopar) [inline doc/input-output-reference/src/overview/group-design-objects.tex:1530]: Ok.

File: src/EnergyPlus/DataHeatBalFanSys.hh [hunk 5]
```diff
@@ -142,8 +142,8 @@ namespace DataHeatBalFanSys {
 	extern Array1D< Real64 > MCPTThermChim; // THERMALCHIMNEY MASS FLOW * AIR SPECIFIC HEAT
 	extern Array1D< Real64 > MCPThermChim; // THERMALCHIMNEY MASS FLOW * AIR CP * AIR TEMPERATURE
 	extern Array1D< Real64 > ZoneLatentGain; // Latent Energy from each Zone (People, equipment)
-	extern Array1D< Real64 > OAMFL; // OUTDOOR AIR MASS FLOW (M**3/SEC) for infiltration
-	extern Array1D< Real64 > VAMFL; // OUTDOOR AIR MASS FLOW (M**3/SEC) for ventilation
+	extern Array1D< Real64 > OAMFL; // OUTDOOR AIR MASS FLOW (kg/s) for infiltration
+	extern Array1D< Real64 > VAMFL; // OUTDOOR AIR MASS FLOW (kg/s) for ventilation
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataHeatBalFanSys.hh:146]: :+1:

File: src/EnergyPlus/DataZoneEquipment.cc [hunk 6]
```diff
@@ -1300,22 +1309,75 @@ namespace DataZoneEquipment {
...
+						int curNodeNum = thisZoneEquip.ReturnNode( nodeCount );
+						if ( NodeName == DataLoopNode::NodeID( curNodeNum ) ) {
+							ReturnAirNodeNumber = curNodeNum;
+						}
+					}
+				}
+			}}
+		}
+
+		return ReturnAirNodeNumber;
```
Comment (Myoldmopar) [inline src/EnergyPlus/DataZoneEquipment.cc:1332]: So, at some point, when the entire codebase is setup to handle the possibility of multiple nodes per zone, this function should never need to warn anymore, right?  The NodeName will be required and provided once the callers all understand the new paradigm.  I hope so, but if I'm wrong then I just want to understand why.

File: src/EnergyPlus/OutputReportTabular.cc [hunk 7]
```diff
@@ -11850,99 +11850,74 @@ namespace OutputReportTabular {
...
-				}
-				if ( displayZoneComponentLoadSummary && ( AirLoopZonesHeatCompLoadTables( iZone ).desDayNum == ZoneHeatCompLoadTables( iZone ).desDayNum ) && ( AirLoopZonesHeatCompLoadTables( iZone ).timeStepMax == ZoneHeatCompLoadTables( iZone ).timeStepMax ) ) {
-					AirLoopZonesHeatCompLoadTables( iZone ) = ZoneHeatCompLoadTables( iZone );
-				} else {
-					heatDesSelected = AirLoopZonesHeatCompLoadTables( iZone ).desDayNum;
-					timeHeatMax = AirLoopZonesHeatCompLoadTables( iZone ).timeStepMax;
+				// now go through the zones and if design day and time of max match the previously calculated zone results use those otherwise compute them for specific design day and time of max
+				for ( int iZone = 1; iZone <= NumOfZones; ++iZone ) {
+					if ( !ZoneEquipConfig( iZone ).IsControlled ) continue;
+					if  ( displayZoneComponentLoadSummary && (AirLoopZonesCoolCompLoadTables( iZone ).desDayNum == ZoneCoolCompLoadTables( iZone ).desDayNum) && (AirLoopZonesCoolCompLoadTables( iZone ).timeStepMax == ZoneCoolCompLoadTables( iZone ).timeStepMax) ){
```
Comment (Myoldmopar) [inline src/EnergyPlus/OutputReportTabular.cc:11896]: A couple whitespace things here; let's see if any more commits are needed.

File: src/EnergyPlus/SimAirServingZones.cc [hunk 8]
```diff
@@ -682,14 +685,17 @@ namespace SimAirServingZones {
 					break;
 				}
 			}
-			if ( ( test == 0 ) && ( AirToZoneNodeInfo(AirSysNum).NumReturnNodes > 0 ) ) { // there, see if it's in the controlled zone info
+			if ( ( test == 0 ) && ( AirToZoneNodeInfo( AirSysNum ).NumReturnNodes > 0 ) ) { // there, see if it's in the controlled zone info
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimAirServingZones.cc:688]: Hmm... `test` doesn't make a great variable name.  But I know you didn't add it, so no worries.

File: src/EnergyPlus/SystemAvailabilityManager.cc [hunk 9]
```diff
@@ -3633,11 +3505,7 @@ namespace SystemAvailabilityManager {
 								CycleOnFlag = true;
 								ATGUpdateFlag2 = true;
 								ATGUpdateFlag1 = true;
-								for ( I = 1; I <= NumOfZones; ++I ) {
-									if ( ZoneEquipConfig( I ).AirLoopNum == PriAirSysNum ) {
-										OptStartData.OptStartFlag( ZoneEquipConfig( I ).ActualZoneNum ) = true;
-									}
-								}
+								OptStartMgr.SetOptStartFlag( PriAirSysNum );
```
Comment (Myoldmopar) [inline src/EnergyPlus/SystemAvailabilityManager.cc:3508]: I just love all these changes so much. :heart:

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 10]
```diff
@@ -4073,6 +3996,145 @@ namespace ZoneEquipmentManager {
...
+		int const ZoneNum,
+		Real64 & ExpTotalReturnMassFlow, // Expected total return air mass flow rate
+		Real64 & FinalTotalReturnMassFlow // Final total return air mass flow rate
+	)
+	{
+		auto & thisZoneEquip( ZoneEquipConfig( ZoneNum ) );
+		int numRetNodes = thisZoneEquip.NumReturnNodes;
+		Real64 totReturnFlow = 0.0; // Total flow to all return nodes in the zone (kg/s)
+		Real64 totVarReturnFlow = 0.0; // Total variable return flow, for return nodes connected to an airloop with an OA system or not with specified flow (kg/s)
+		Real64 returnSchedFrac = ScheduleManager::GetCurrentScheduleValue( thisZoneEquip.ReturnFlowSchedPtrNum );
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:4029]: You feel OK calling this without validation of the index variable?  I'm sure it's done elsewhere, so it's probably fine...

File: src/EnergyPlus/ZoneEquipmentManager.cc [hunk 11]
```diff
@@ -4073,6 +3996,145 @@ namespace ZoneEquipmentManager {
...
+				if ( retNode > 0 ) {
+					newReturnFlow = curReturnFlow * returnAdjFactor;
+					FinalTotalReturnMassFlow += newReturnFlow;
+					DataLoopNode::Node( retNode ).MassFlowRate = newReturnFlow;
+				}
+			}
+		} else {
+			FinalTotalReturnMassFlow = totReturnFlow;
+		}
+
```
Comment (Myoldmopar) [inline src/EnergyPlus/ZoneEquipmentManager.cc:4137]: Whew, quite a function :+1:

File: tst/EnergyPlus/unit/HVACFourPipeBeam.unit.cc [hunk 12]
```diff
@@ -1638,6 +1639,7 @@ namespace EnergyPlus {
 		DataGlobals::KickOffSimulation = true;
 
 		WeatherManager::ResetEnvironmentCounter();
+		TestAirPathIntegrity( ErrorsFound ); // Needed to initialize return node connections to airloops and inlet nodes
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/HVACFourPipeBeam.unit.cc:1642]: :+1:

File: tst/EnergyPlus/unit/ThermalComfort.unit.cc [hunk 13]
```diff
@@ -390,7 +390,7 @@ TEST_F( EnergyPlusFixture, ThermalComfort_CalcThermalComfortFanger )
 		" ,                         !- Lighting Level{ W }",
 		" 10.0,                     !- Watts per Zone Floor Area{ W / m2 }",
 		" ,                         !- Watts per Person{ W / person }",
-		" 0.1,                      !- Return Air Fraction",
+		" 0.0,                      !- Return Air Fraction",
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ThermalComfort.unit.cc:393]: Is this to avoid having to change all the values being checked by the unit test?

File: tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc [hunk 14]
```diff
@@ -437,3 +444,140 @@ TEST_F( EnergyPlusFixture, ZoneEquipmentManager_MultiCrossMixingTest )
...
+	Node( returnNode3 ).MassFlowRate = 0.45;
+
+	StdTotalReturnMassFlow = 0.0;
+	FinalTotalReturnMassFlow = 0.0;
+
+	CalcZoneReturnFlows( ZoneNum, StdTotalReturnMassFlow, FinalTotalReturnMassFlow );
+	EXPECT_EQ( FinalTotalReturnMassFlow, 6.0 );
+	EXPECT_EQ( Node( returnNode1 ).MassFlowRate, 2.0 );
+	EXPECT_EQ( Node( returnNode2 ).MassFlowRate, 1.0 );
+	EXPECT_EQ( Node( returnNode3 ).MassFlowRate, 3.0 );
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/ZoneEquipmentManager.unit.cc:579]: :+1: :+1: :+1:

**Review by Myoldmopar — APPROVED** [review]
This is **exactly** what I was expecting to find.  Tons and tons of bookkeeping.  Changing single lines of code where nodes where looked up or assigned into loops.  This is a huge amount of code changes, but they look good.  It's great to see some tests added on to existing unit tests, and the new unit tests look like they cover core functionality.  The doc changes look good, and they build locally for me.  I'll check them out visually and do some testing, but I think this is good to go.

**Comment by Myoldmopar** [comment]
@mjwitte Apparently I had started a review a while back and tried to comment to you about the REGRESSION_SHA in the CMakeLists file.  I guess Github kept that comment in a cache, even though that diff is long gone.  Anyway, sorry for any confusion on the CMakeLists.txt (outdated) comment above.

File: doc/input-output-reference/src/overview/group-design-objects.tex (hunk 3 above)
Comment (mjwitte) [reply doc/input-output-reference/src/overview/group-design-objects.tex:887]: Whacky - you know when I made the doc changes last night, the local commit diff showed this line and another even though I hadn't changed anything, so I thought maybe my editor had automatically done something it shouldn't.  So, I undid those before I pushed it up. <sigh>

**Comment by Myoldmopar** [comment]
You don't think this bnd diff is going to cause a problem in HVACDiagram or other tools do you?

image

**Comment by Myoldmopar** [comment]
Interesting.  Checking out the diffs on 5ZoneAirCooled_ZoneAirMassFlowBalance, and the EIO is showing better convergence at the end of the warmup days.  Nothing significant, but to be honest, I kinda expected to see the system having a bit more freedom and taking a little longer to converge.  Looks like this improved it.  Or it was just luck?

(new EIO lines on the bottom in blue)

image

**Comment by Myoldmopar** [comment]
OK, if no one else has anything to say on this I'm merging it in very soon.

File: src/EnergyPlus/DataZoneEquipment.cc (hunk 6 above)
Comment (mjwitte) [reply src/EnergyPlus/DataZoneEquipment.cc:1332]: Good question.  I had wanted the node names to be optional in places like return air heat gain from lights so if you do have only one return node, you don't have to enter a node name.  If we make the return node required, then it wouldn't work with a zonelist name anymore.  So, not sure.

Comment (Myoldmopar) [reply src/EnergyPlus/DataZoneEquipment.cc:1332]: That is interesting.  And tricky.  I see your point, but it feels funny as optional.  Nothing to worry about for today, that's for sure.  We'll see how it plays out.

File: tst/EnergyPlus/unit/ThermalComfort.unit.cc (hunk 13 above)
Comment (mjwitte) [reply tst/EnergyPlus/unit/ThermalComfort.unit.cc:393]: Return air doesn't impact thermal comfort calcs. This change was to avoid tripping on [this error](https://github.com/NREL/EnergyPlus/blob/fc306778e169208ee5d29ce9d678d1e57f13eb53/src/EnergyPlus/InternalHeatGains.cc#L1152-L1156) because the unit test doesn't have any conditioned zone info.

**Comment by Myoldmopar** [comment]
> You don't think this bnd diff is going to cause a problem in HVACDiagram or other tools do you?

Oh I guess #6345 explains it :laughing:

**Comment by mjwitte** [comment]
The bnd change doesn't affect HVAC diagram, not sure if that's used by any other tools.  I can revert that back and save it for post-release.  But that reminds me, HVAC diagram chokes on zones with more than one airloop (e.g. 5ZoneAirCooledWithDOASAirLoop ).  Posted #6345 for that.

Regarding 5ZoneAirCooled_ZoneAirMassFlowBalance, the mass balance was broken before.  It works now - actually balances.

**Comment by Myoldmopar** [comment]
OK, thanks for all these comments.

**Comment by mjwitte** [comment]
Thanks for the review.  I can clean up the funny doc chars and the whitespace stuff if you want.  Or just take care of them in that other open PR.

**Comment by Myoldmopar** [comment]
"Documentation Issues October-November 2017" :+1:

File: src/EnergyPlus/OutputReportTabular.cc (hunk 7 above)
Comment (mjwitte) [reply src/EnergyPlus/OutputReportTabular.cc:11896]: Fixed this over ini #6338 .
